### ISSUE TITLE
Don't break shields unless the damage will kill you

### ISSUE THREAD

**Issue by winstliu** [50DKP/FF2-Official#150]
This is helpful in a number of ways (for example, fire attacks and the Chargin' Targe), and further justifies the default removal of demoshield crits.

**Comment by Bonkorn** [comment]
Just found this from my code.
`if(shield[client] && damage)
			{
				new health=GetClientHealth(client);
				if(health<210) // shield breaks when below 210HP
				{
					RemoveShield(client, attacker, position);
				}
				else if(damage>210) // shield breaks when receive above 210 damage like GentleSpy's ambassador even above 210HP
				{
					RemoveShield(client, attacker, position);
				}
				else if(health>210) // no shield breaks when above 210HP
				{
					return Plugin_Changed;
				}
				return Plugin_Handled;
			}`

**Comment by Bonkorn** [comment]
This?
			if(shield[client] && damage)
			{
				new health=GetClientHealth(client);
				if(health<210) // shield breaks when below 210HP
				{
					RemoveShield(client, attacker, position);
				}
				else if(damage>210) // shield breaks when receive above 210 damage like GentleSpy's ambassador even above 210HP
				{
					RemoveShield(client, attacker, position);
				}
				else if(health>210) // no shield breaks when above 210HP
				{
					return Plugin_Changed;
				}
				return Plugin_Handled;
			}

**Comment by Bonkorn** [comment]
if(shield[client] && damage)
			{
				new health=GetClientHealth(client);
				if(health<210) // shield breaks when below 210HP
				{
					RemoveShield(client, attacker, position);
				}
				else if(damage>210) // shield breaks when receive above 210 damage like GentleSpy's ambassador even above 210HP
				{
					RemoveShield(client, attacker, position);
				}
				else if(health>210) // no shield breaks when above 210HP
				{
					return Plugin_Changed;
				}
				return Plugin_Handled;
			}

This?

**Comment by winstliu** [comment]
Not necessarily, because even a 1-damage tick will remove your shield if you're < 210 health.

**Comment by Bonkorn** [comment]
if(shield[client] && damage)
			{
				new health=GetClientHealth(client);
				if(health<damage)
				{
					RemoveShield(client, attacker, position);
				}
				else if(health>damage)
				{
					return Plugin_Changed;
				}
				return Plugin_Handled;
			}
Maybe this better.
